packages/fxa-auth-server/lib/payments/configuration/base.ts

```diff
@@ -0,0 +1,107 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import joi from '@hapi/joi';
+
+export type CapabilityConfig = {
+  [key: string]: string;
+};
+
+const capabilitySchema = joi
+  .object()
+  .pattern(joi.string(), joi.string().min(1))
+  .min(1);
+
+export type UrlConfig = {
+  download?: string;
+  webIcon?: string;
+  emailIcon?: string;
+  termsOfService?: string;
+  termsOfServiceDownload?: string;
+  privacyNotice?: string;
+  privacyNoticeDownload?: string;
+  playStore?: string;
+  appStore?: string;
+};
+
+const urlsSchema = joi.object({
+  download: joi.string().uri(),
+  webIcon: joi.string().uri(),
+  emailIcon: joi.string().uri(),
+  termsOfService: joi.string().uri(),
+  termsOfServiceDownload: joi.string().uri(),
+  privacyNotice: joi.string().uri(),
+  privacyNoticeDownload: joi.string().uri(),
```

Comment:
According to our [existing Joi product validator](https://github.com/mozilla/fxa/blob/f32ae3f1a02969766a7ba4686907f941793a8fa6/packages/fxa-auth-server/lib/routes/validators.js#L429), [termsOfServiceDownload](https://github.com/mozilla/fxa/blob/f32ae3f1a02969766a7ba4686907f941793a8fa6/packages/fxa-auth-server/lib/routes/validators.js#L442-L445) and [privacyNoticeDownload](https://github.com/mozilla/fxa/blob/f32ae3f1a02969766a7ba4686907f941793a8fa6/packages/fxa-auth-server/lib/routes/validators.js#L447-L450) should both require the FxA CDN legal resource domain.

I am not sure what the original motivation for that requirement was -- @chenba should we enforce that here as well?

Reply:
The "motivation" is from the legal docs redirect that handles l10n won't redirect unless the domain matches the configured domain, which by default is the FxA CDN domain.  Definitely couldn't hurt to enforce it here (unless there are testing scenarios that would be negatively affected).

Reply:
I could add that here, but I'd anticipate this patch getting larger yet to handle that. This value would be contextually dependent and require a validator that does something like `startsWith`, which joi doesn't gracefully handle. I think we'll need to screen such values higher up the stack, or with additional custom validation logic that runs before this based on config values.

A separate ticket might be good for that, when we're closer to having an endpoint that takes front-end submitted objects.